app/controllers/api/entries/icca_data_entries_controller.rb

```diff
@@ -21,7 +21,8 @@ def all_entries
     if endpoint_return['status_code'] != REQUESTSTATUSTYPE['ok']['code']
       return render json: endpoint_return, status: endpoint_return['status_code']
     end
-    type = processed_params['type'] ? processed_params['type'] : 'file_export'
+
+    type = processed_params['type'] || 'file_export'
```

Comment:
```suggestion
    type = processed_params['type'].presence || 'file_export'
```
Here we can add presence just to handle the scenarios where processed_params['type'] will be an empty string or a string with spaces only.

Reply:
thanks for the tips. in this scenario it will be caught by the line 17 if the value doesn't belong to any of queries_allowed_to_be_used_externally key but I have added as it is a good approach